### DISCUSSION TITLE
Components aren't fetched again after loaded once

### DISCUSSION THREAD

**Discussion by nelhwong** [FranckFreiburger/vue3-sfc-loader#12] [Q&A] [answered]
Hi, I'm new to Vue and I'm just experimenting with a Flask backend.

I've noticed that a component gets fetched when I first use the "loadModule" method on the particular file, however when I reload the page, the components aren't getting fetched again. I've add the cache control headers in the html file and flask. I think this has to do with "moduleCache". If there's any way I can clear the cache or force fetch, it would be great!

I've also looked into the issues from http-vue-loader. [https://github.com/FranckFreiburger/http-vue-loader/issues/109#issuecomment-615577749](url)
Apparently adding a hash at the end works, but it work wont for this because it blocks the hash in index.tsx:710
`const defaultModuleHandlers : Record<string, ModuleHandler> = {
	'.vue': (source, path, options) => createSFCModule(source, path, options),
	'.js': (source, path, options) => createJSModule(source, false, path, options),
	'.mjs': (source, path, options) => createJSModule(source, true, path, options),
};`

Thanks and happy new year 😊

**Comment by FranckFreiburger** [comment, marked as answer]
Hi and happy new year !

Your `getFile()` may return a file content or an object `{ content, extname }`

src: https://github.com/FranckFreiburger/vue3-sfc-loader/blob/d5ece4483c24c51b64dfa2f7061f39992344e3c2/src/index.ts#L778-L780
doc: https://github.com/FranckFreiburger/vue3-sfc-loader/blob/1dee31300c675f2b232d825ed81fb2730aad0e90/docs/api/README.md#file

This can allow you to fetch any resource (maybe with hash) and set `extname` to the right extension.
